plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -110,9 +111,19 @@ public DockerInstanceProvider(DockerConnector docker,
 
         for (ServerConf serverConf : allMachineServers) {
             portsToExposeOnMachine.put(serverConf.getPort(), Collections.<String, String>emptyMap());
+            portsToExposeOnDevMachine.put(serverConf.getPort(), Collections.<String, String>emptyMap());
             machineContainerLabels.put("che:server:" + serverConf.getPort() + ":ref", serverConf.getRef());
+            devMachineContainerLabels.put("che:server:" + serverConf.getPort() + ":ref", serverConf.getRef());
             machineContainerLabels.put("che:server:" + serverConf.getPort() + ":protocol", serverConf.getProtocol());
+            devMachineContainerLabels.put("che:server:" + serverConf.getPort() + ":protocol", serverConf.getProtocol());
         }
+
+        this.systemVolumesForDevMachine.addAll(allMachinesSystemVolumes);
+        this.systemVolumesForDevMachine.addAll(systemVolumesForDevMachine);
+
+        commonEnvVariables = new String[0];
```

Comment:
It looks like <code>commonEnvVariables</code> is a constant, so maybe it should be declared as <code>private static final</code> field ?